src/clj_momo/lib/es/document.clj

```diff
@@ -304,32 +313,65 @@
     mapping :- (s/maybe s/Str)]
    (count-docs es-conn index-name mapping nil)))
 
-(s/defn query
-  "Search for documents on ES using any query."
+(s/defn scroll
```

Comment:
The diff show that `scroll` and `query` functions share a lot of code in common.
Perhaps there is a way to prevent duplication. Just take a look, I generally prefer duplication than creating an intermediate function with too much parameters.

But just looking at the diff, it looks like a good candidate to find a perhaps more generic/powerful function for both `scroll` and `query`.

Reply:
We chosen with @gbuisson to preserve the Elasticsearch way: [query then scroll](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-request-scroll.html).
I can find a way to factorize the parsing of ES response.

Reply:
ES may send you a new one when the scroll id is about to expire, but it will send you the same one if it is not.
I have no clue on how know how to simulate that. 
If it expires before the next scroll request (with a sleep between two calls for instance), the next scroll request will return empty data and empty scroll_id because ES would have deleted the corresponding search context.